### ISSUE TITLE
Crashes when Android 14 Memory Tagging feature is enabled.

### ISSUE THREAD

**Issue by hesoyam-enjoyer** [osmandapp/OsmAnd#19062]
### Description

App immediately crashes when launched on the OS that supports Memory Tagging
Logs of the crash are provided bellow.

### Steps to reproduce

Enable Memory Tagging (GrapheneOS -> Settings -> Security Settings -> Memory Tagging)
Download any version of OsmAnd (from any source, Google Play, FDroid, etc.)
Launch an app

### Actual result

App immediately crashes to desktop every single launch.

### Expected result

App should just open and work as regularly.

### Your Environment (required)

```
OsmAnd Version: 4.6.13
Android/iOS version: GrapheneOS (build 2024020500, base os - Android AOSP 14, UQ1A.240205.004)
Device model: Google Pixel 8
Crash-Logs:

type: crash
osVersion: google/husky/husky:14/UQ1A.240205.004/2024020500:user/release-keys
uid: 10150 (u:r:untrusted_app_32:s0:c150,c256,c512,c768)
cmdline: net.osmand.plus
processUptime: 4s

signal: 11 (SIGSEGV), code 9 (SEGV_MTESERR), faultAddr 200cd9f41018200
threadName: GLThread 39
MTE: enabled

backtrace:
    /apex/com.android.runtime/lib64/bionic/libc.so (__strlen_aarch64_mte+8, pc 5bf08)
    /apex/com.android.runtime/lib64/bionic/libc.so (__strlen_chk+16, pc 91c60)
    /data/app/~~GfNC8iyGifgAh_RFvgVyXg==/net.osmand.plus-8MMLAY5yY5qK6tOXHKD6LA==/split_config.arm64_v8a.apk!libOsmAndCoreWithJNI.so (OsmAnd::GPUAPI_OpenGL::linkProgram(unsigned int, unsigned int const*, QList<std::__ndk1::tuple<OsmAnd::GlslVariableType, QString, int> > const&, bool, QHash<QString, OsmAnd::GPUAPI_OpenGL::GlslProgramVariable>*)+1480, pc ad6fac)
    /data/app/~~GfNC8iyGifgAh_RFvgVyXg==/net.osmand.plus-8MMLAY5yY5qK6tOXHKD6LA==/split_config.arm64_v8a.apk!libOsmAndCoreWithJNI.so (OsmAnd::GPUAPI_OpenGL::linkProgram(unsigned int, unsigned int const*, bool, QHash<QString, OsmAnd::GPUAPI_OpenGL::GlslProgramVariable>*)+60, pc ad6998)
    /data/app/~~GfNC8iyGifgAh_RFvgVyXg==/net.osmand.plus-8MMLAY5yY5qK6tOXHKD6LA==/split_config.arm64_v8a.apk!libOsmAndCoreWithJNI.so (OsmAnd::AtlasMapRendererSkyStage_OpenGL::initialize()+820, pc aaede0)
    /data/app/~~GfNC8iyGifgAh_RFvgVyXg==/net.osmand.plus-8MMLAY5yY5qK6tOXHKD6LA==/split_config.arm64_v8a.apk!libOsmAndCoreWithJNI.so (OsmAnd::AtlasMapRenderer::doInitializeRendering()+36, pc a89310)
    /data/app/~~GfNC8iyGifgAh_RFvgVyXg==/net.osmand.plus-8MMLAY5yY5qK6tOXHKD6LA==/split_config.arm64_v8a.apk!libOsmAndCoreWithJNI.so (OsmAnd::AtlasMapRenderer_OpenGL::doInitializeRendering()+16, pc a310a0)
    /data/app/~~GfNC8iyGifgAh_RFvgVyXg==/net.osmand.plus-8MMLAY5yY5qK6tOXHKD6LA==/split_config.arm64_v8a.apk!libOsmAndCoreWithJNI.so (OsmAnd::MapRenderer::initializeRendering(bool)+96, pc a40014)
    /data/app/~~GfNC8iyGifgAh_RFvgVyXg==/net.osmand.plus-8MMLAY5yY5qK6tOXHKD6LA==/split_config.arm64_v8a.apk!libOsmAndCoreWithJNI.so (Java_net_osmand_core_jni_OsmAndCoreJNI_IMapRenderer_1initializeRendering+40, pc 741fe4)
    /data/app/~~GfNC8iyGifgAh_RFvgVyXg==/net.osmand.plus-8MMLAY5yY5qK6tOXHKD6LA==/oat/arm64/base.odex (art_jni_trampoline+136, pc 2b353b8)
    /data/app/~~GfNC8iyGifgAh_RFvgVyXg==/net.osmand.plus-8MMLAY5yY5qK6tOXHKD6LA==/oat/arm64/base.odex (net.osmand.core.jni.IMapRenderer.initializeRendering+52, pc 2af0794)
    /data/app/~~GfNC8iyGifgAh_RFvgVyXg==/net.osmand.plus-8MMLAY5yY5qK6tOXHKD6LA==/oat/arm64/base.odex (net.osmand.core.android.MapRendererView$RendererProxy.onSurfaceChanged+696, pc 134db08)
    /apex/com.android.art/lib64/libart.so (nterp_helper+7636, pc 20b154)
    /system/framework/framework.jar (android.opengl.GLSurfaceView$GLThread.guardedRun+1000, pc 4df894)
    /apex/com.android.art/lib64/libart.so (nterp_helper+3924, pc 20a2d4)
    /system/framework/framework.jar (android.opengl.GLSurfaceView$GLThread.run+52, pc 4dff2c)
    /apex/com.android.art/lib64/libart.so (art_quick_invoke_stub+612, pc 2109a4)
    /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+172, pc 253b3c)
    /apex/com.android.art/lib64/libart.so (art::Thread::CreateCallback(void*)+1416, pc 69abc8)
    /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+204, pc d006c)
    /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64, pc 64db0)
```

**Comment by yuriiurshuliak** [comment]
Duplicate of #18659
The latest release version already has a fix.

**Comment by hesoyam-enjoyer** [comment]
@yuriiurshuliak Well, latest release available in Google Play Store (4.6.13) is the version I tested today and got this error. 
Is there some newer release version somewhere?

**Comment by DmitryAlexei** [comment]
@hesoyam-enjoyer, please join beta from Google play. A quick fix, that removes the crash, but still doesn't solve the initial issue, should be already available.
